corehq/apps/export/export.py

```diff
@@ -336,19 +310,19 @@ def get_export_file(export_instances, filters, progress_tracker=None):
 def get_export_documents(export_instance, filters):
     # Pull doc ids from elasticsearch and stream to disk
     query = _get_export_query(export_instance, filters)
-    _, temp_path = tempfile.mkstemp()
-    with open(temp_path, 'w') as f:
-        scroll_result = query.scroll_ids()
-        for doc_id in scroll_result:
-            f.write(doc_id + '\n')
+    scroll_result = query.scroll_ids()
 
     def iter_export_docs():
-        # Stream doc ids from disk and fetch documents from ES in chunks
-        with open(temp_path) as f:
-            doc_ids = (doc_id.strip() for doc_id in f)
-            for doc in iter_es_docs(query.index, doc_ids):
-                yield doc
-        os.remove(temp_path)
+        with TransientTempfile() as temp_path:
+            with open(temp_path, 'w', encoding='utf-8') as f:
+                for doc_id in scroll_result:
+                    f.write(doc_id + '\n')
```

Comment:
would it be better to write at once?
`f.write('\n'.join(doc_id for doc_id in scroll_result))`

This would probably depend on the number of ids here. if its just a few or too large this might be not-much-difference/too-much-memory but does avoid that many `write` method calls.

Reply:
It's potentially a very big list, and I think that approach would require pulling it all into memory and committing in one go.  Perhaps a chunked approach would be better?  I'll call that out of scope for this PR, since I wasn't intending to change what happens with the tempfiles, just that they are guaranteed to get cleaned up.